opnode/withdrawals/proof.go

```diff
@@ -0,0 +1,100 @@
+package withdrawals
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient/gethclient"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/trie"
+)
+
+type proofDB struct {
```

Comment:
For the output root construction we've some proof code already. But without the storage proof part, just the account proof. There it types the rlp nodes as `hexutil.Bytes` and maintains a map of `common.Hash` to nodes, instead of using strings for keys. Here it nicely uses the account type + some of the geth rpc client types though. In both cases it does verify correctly, and does verify all the things in the response (i.e. the account details in the response are checked against the rlp node in the proof). Either way, some opportunity for a later PR to take the best of both and reduce code.